src/fmt.rs

```diff
@@ -1,19 +1,52 @@
 use std::fmt::Write as _;
 
-pub(crate) fn autoformat_leading(leading: &mut String, indent: usize, no_comments: bool) {
+/// Formatting configuration for use with [`KdlDocument::autoformat_config`](`crate::KdlDocument::autoformat_config`)
+/// and [`KdlNode::autoformat_config`](`crate::KdlNode::autoformat_config`).
+#[non_exhaustive]
+#[derive(Debug)]
+pub struct FormatConfig<'a> {
+    /// How deeply to indent the overall node or document,
+    /// in repetitions of [`indent`](`FormatConfig::indent`).
+    /// Defaults to `0`.
+    pub indent_level: usize,
+
+    /// The indentation to use at each level. Defaults to four spaces.
+    pub indent: &'a str,
```

Comment:
I went with just `indent` for the field name here, and a `&str` for repetitions of the same character.

Reply:
Alternatively, this could be an enum of `Tab` and `Spaces(usize)` to constrain it to valid options 🤔